tools/toolchains/arm.py

```diff
@@ -81,6 +92,9 @@ def __init__(self, target, notify=None, macros=None,
             if "--library_type=microlib" not in self.flags['common']:
                 self.flags['common'].append("--library_type=microlib")
 
+        if getattr(target, "default_lib", "std") == "small" and getattr(target, "ulib_supported", False) == True:
+            self.flags['ld'].append("--predefine=-D__MICROLIB")
```

Comment:
You should include as part of the if statement above and follow the same pattern
```
if "--predefine=-D__MICROLIB" not in self.flags['ld']:
   self.flags['ld'].append("--predefine=-D__MICROLIB")
```

Reply:
We need to specify linker predefine with __MICROLIB to enable microlib related content in scatter file and this linker configuration is not required for uARM toolchain

Reply:
If you want to avoid the flag being added for uARM toolchain then the check is not correct as you would need:
```
if getattr(target, "default_toolchain", "ARM") != "uARM" and getattr(target, "default_lib", "std") == "small"
```
As mentioned in my comment, you should also check 
```
if "--predefine=-D__MICROLIB" not in self.flags['ld']:
```
I don' t think it hurts to add the flag for uARM as well so I would go with my earlier suggestion unless tools team disagrees. I don't have a strong opinion on this.